### PR TITLE
SIMSBIOHUB-8: Validation/Transformation Updates/Cleanup

### PR DESCRIPTION
**Pull request by NickPhura**
# Overview

## Links to Jira tickets

Not directly related to SIMSBIOHUB-8, but some things noticed while working in the biohubbc-utils repo.

## Description of relevant changes

- Remove unused `notifications.ts` const.
- Cleaned up the validation/transformation json-schema definitions, as they were a bit out of date. 
- removed `xlsx-transform-json-path-queries.ts` as it is a helper file used in building configs (used in biohubbc-utils) but never used in SIMS.
- Updated a bunch of `object` types to be `Record<string,any>`. 
- Added more comments/JSDoc to validation/transformation functions/types.
- Added more sophisticated support for transforming date and time cells.

## PR Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

### CODE REVIEW

File: api/package.json [hunk 1]
```diff
@@ -59,7 +60,7 @@
     "utm": "^1.1.1",
     "uuid": "~8.3.2",
     "winston": "~3.3.3",
-    "xlsx": "~0.18.5",
+    "xlsx": "https://cdn.sheetjs.com/xlsx-0.19.3/xlsx-0.19.3.tgz",
```
Comment (NickPhura) [inline api/package.json:63]: xlsx (aka sheetjs) is no longer hosting any of their new releases on npm

File: api/src/utils/media/xlsx/transformation/xlsx-transform.ts [hunk 2]
```diff
@@ -178,6 +230,7 @@ export class XLSXTransform {
         return RowObject[columnName];
       })
       .filter(Boolean)
+      .filter(Boolean)
```
Comment (curtisupshall) [inline api/src/utils/media/xlsx/transformation/xlsx-transform.ts]: Is the double `.filter(Boolean)` intentional?

**Review by curtisupshall — DISMISSED** [review]
Looks great! +1 for all the documentation added to all the validators 😎

**Review by anissa-agahchen — DISMISSED** [review]

File: api/src/utils/media/xlsx/transformation/xlsx-transform.ts (hunk 2 above)
Comment (NickPhura) [reply api/src/utils/media/xlsx/transformation/xlsx-transform.ts]: Nope! I think I had changes across 2 PRs, so probably just a funny merge. Will fix.

Comment (NickPhura) [reply api/src/utils/media/xlsx/transformation/xlsx-transform.ts]: Fixed

**Review by anissa-agahchen — APPROVED** [review]

**Review by al-rosenthal — APPROVED** [review]
🍏

**Pull request merged**